src/core/splitter/utils.js

```diff
@@ -0,0 +1,287 @@
+
+		target.x = ( x1 + x2 ) / 2;
+		target.y = ( y1 + y2 ) / 2;
+		return true;
+
+	}
+
+	/* Are the line parallel */
+	if ( Math.abs( denom ) < EPS ) {
+
+		target.x = 0;
+		target.y = 0;
+		return false;
+
+	}
+
+	/* Is the intersection along the the segments */
+	mua = numera / denom;
+	mub = numerb / denom;
+	if ( mua < 0 || mua > 1 || mub < 0 || mub > 1 ) {
+
+		target.x = 0;
+		target.y = 0;
+
+		return false;
+
+	}
+
+	target.x = x1 + mua * ( x2 - x1 );
+	target.y = y1 + mua * ( y2 - y1 );
+
+	return true;
+
+}
+
+export function getTriangleLineIntersection( line, tri, target ) {
+
+	let setCount = 0;
+	const vec = new Vector3();
+	const edge = new Line3();
+	const arr = [ tri.a, tri.b, tri.c ];
+	const EPS = 1e-10;
+	for ( let i = 0; i < 3; i ++ ) {
+
+		const ni = ( i + 1 ) % 3;
+		edge.start.copy( arr[ i ] );
+		edge.end.copy( arr[ ni ] );
+
+		if ( areEdgesParallel( edge, line ) ) {
+
+			// TODO: we need to check if the lines are actually on each other
+			edge.closestPointToPoint( line.start, false, vec );
+			if ( line.start.distanceTo( vec ) < EPS ) {
+
+				let sp = edge.closestPointToPointParameter( line.start, false );
+				let ep = edge.closestPointToPointParameter( line.end, false );
+				if ( ! (
+					sp < 0 && ep < 0 ||
+					sp > 1 && ep > 1
+				) ) {
+
+					sp = MathUtils.clamp( sp, 0, 1 );
+					ep = MathUtils.clamp( ep, 0, 1 );
+
+					edge.at( sp, target.start );
+					edge.at( ep, target.end );
+
+					return true;
+
+				}
+
+			}
+
+			continue;
+
+		} else if ( lineIntersect( edge, line, vec ) ) {
+
+			if ( setCount === 2 ) {
+
+				// TODO
+				const d0 = target.start.distanceTo( vec );
+				const d1 = target.end.distanceTo( vec );
+				if ( d0 > EPS && d1 > EPS ) {
+
+					console.error( 'This shouldn\'t happen' );
+
+				}
+
+			} else if ( setCount === 1 ) {
+
+				if ( vec.distanceTo( target.end ) > EPS ) {
+
+					target.start.copy( vec );
+					setCount ++;
+
+				}
+
+			} else {
+
+				target.end.copy( vec );
+				setCount ++;
+
+			}
+
+		}
+
+	}
+
+	if ( setCount === 0 || setCount === 1 ) {
+
+		const cs = tri.containsPoint( line.start );
+		const ce = tri.containsPoint( line.end );
+
+		if ( setCount === 0 ) {
+
+			if ( cs && ce ) {
+
+				target.copy( line );
+				setCount += 2;
+
+			} else if ( cs || ce ) {
+
+				// TODO
+				console.error( 'This shouldn\'t happen' );
+
+			}
+
+		}
+
+		if ( setCount === 1 ) {
+
+			if ( cs && ce ) {
+
+				if ( line.start.distanceTo( target.end ) > line.end.distanceTo( target.end ) ) {
+
+					target.start.copy( line.start );
+
+				} else {
+
+					target.start.copy( line.end );
+
+				}
+
+				setCount ++;
+
+			} else if ( cs ) {
+
+				target.start.copy( line.start );
+				setCount ++;
+
+			} else if ( ce ) {
+
+				target.start.copy( line.end );
+				setCount ++;
+
+			}
+
+		}
+
+	}
+
+	// if the final edge is degenerate then we haven't intersected
+	if ( target.distance() < EPS ) {
```

Comment:
I have been trying to get this to work for us locally and I believe this will fail if the triangles only intersect at a single point:

```
__
|  /
| /*----
|/ |
   |
```
Since they are intersecting at that point you still need to insert it into your graph, otherwise the graph will not know about that point

This logic then needs to be moved to the two places you call getTriangleLineIntersection and choose to add a point if it is a single point edge.

Reply:
Causes the issue in the fiddle here:
https://github.com/gkjohnson/three-bvh-csg/issues/210